### ISSUE TITLE
Can't use state value inside successCallback

### ISSUE THREAD

**Issue by ManuDoni** [tomliangg/react-hook-recaptcha#1]
Hi @tomliangg,
I'm trying to use a successCallback that read some state values but they are always old.

This demonstrates the problem:
https://codesandbox.io/s/userecaptcha-example-using-with-react-hook-form-forked-hgbpv?file=/src/App.js

How can I update the recaptcha component with a new successCallback so that I will be able to read the correct state value?

Thanks in advance

**Comment by tomliangg** [comment]
@ManuDoni 
I'd rather say it's a react problem.

Your problem sounds something like "states not updating when using react hook". I found this link which explains the problem perfectly. You may want to give it a read :)
https://stackoverflow.com/questions/53024496/state-not-updating-when-using-react-state-hook-within-setinterval

TL;DR, use `useRef` instead

**Comment by tomliangg** [comment]
@ManuDoni 

I forked your sandbox and made a working example for you
https://codesandbox.io/s/userecaptcha-example-using-with-react-hook-form-forked-k8ghc?file=/src/App.js

This SO post might be relevant for you as well: https://stackoverflow.com/questions/57847594/react-hooks-accessing-up-to-date-state-from-within-a-callback

**Comment by ManuDoni** [comment]
Thank you for the article, it helped me to find a solution, but your sandbox doesn't work.
I created a sandbox that uses my custom hook: https://codesandbox.io/s/react-hook-form-and-react-hook-recaptcha-inside-a-custom-hook-w0qqn?file=/src/App.js
I used a reference to save the onSubmit function and it works but now I have another problem (cannot replicate it in the sandbox).
I have a form with a checkbox that is used to show some more form fields.
After some submission (with server errors) and toggling with the checkbox the form submission stops working.
The `handleSumbit` function gets called but not the `onSubmit` passed as a parameter, I don't know why.

**Comment by tomliangg** [comment]
ohh. Looks like I pasted the wrong link. I'm glad that you found a solution. I can visualize your problem. but without the codes, I can't tell where causes the problems. And your question seems to better fit in Stack Overflow as it's about "how to do XXX with React". I am closing this thread now.
